src/app/pages/ArticlePage/experimentTopStories/helpers.tsx

```diff
@@ -66,23 +68,52 @@ const enableExperimentTopStories = ({
   );
 };
 
+const insertBlockAtPosition = (
+  blocks: OptimoBlock[],
+  blockToInsert: OptimoBlock,
+  position: 'Quarter' | 'Half' | 'ThreeQuarters',
+) => {
+  const insertionPercentages = {
+    Quarter: 0.25,
+    Half: 0.5,
+    ThreeQuarters: 0.75,
+  };
+  const percentage = insertionPercentages[position];
+
+  const calculatedIndex = Math.floor((blocks.length - 1) * percentage); // -1 accounts for 'wsoj' block which is never rendered on PS articles
+  const insertIndex = Math.max(calculatedIndex, 3); // Ensure insertIndex is at least 3
+  const blocksClone = [...blocks];
+  blocksClone.splice(insertIndex, 0, blockToInsert);
+  return blocksClone;
+};
+
 const insertExperimentTopStories = ({
   blocks,
   topStoriesContent,
 }: {
   blocks: OptimoBlock[];
   topStoriesContent: TopStoryItem[];
 }) => {
-  const insertIndex = Math.floor((blocks.length - 1) * 0.5); // halfway index of blocks array, -1 accounts for 'wsoj' block which is never rendered on PS articles
-  const experimentTopStoriesBlock = {
-    type: 'experimentTopStories',
-    model: topStoriesContent,
-    id: `experimentTopStories-${insertIndex}`,
-  };
+  const insertionPositions: ['Quarter', 'Half', 'ThreeQuarters'] = [
+    'Quarter',
+    'Half',
+    'ThreeQuarters',
+  ];
```

Comment:
Used a union type consisting only of what the variants can be rather than just `string`. Just think it looks a bit nicer than typing the `insertionPercentages` object with something like this:
```
insertionPercentages: { [key: string]: number } = { Quarter: 0.25, Half: 0.5, ThreeQuarters: 0.75 }
```

Reply:
This has the same effect with out declaring values twice ([const assertion docs](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions)):
```suggestion
  const insertionPositions = [
    'Quarter',
    'Half',
    'ThreeQuarters',
  ] as const;
```

If you move this up to the top level scope too, then in the `insertBlockAtPosition()` function args typing you can put:
```
position: typeof insertionPositions[number],
```

Then you've only had to type out the possible values once :)